### ISSUE TITLE
How to implement a automatic reconnect attempt to unreachable server

### ISSUE THREAD

**Issue by vtager** [eclipse-paho/paho.mqtt.java#825]
Hi,
Automatically reconnect feature works only if established connection was broken. When I try to connect to unreachable server reconnect doesn't work. Does paho client has support this case?

**Comment by sp193** [comment]
I don't think it does. From my experience, it never attempts to reconnect if the initial connection fails.

If you want it to reconnect, you could call connect() again once the connection fails. If you are using the synchronous client, MqttException is likely thrown from connect(). If you are using the asynchronous client, the onFailure() callback for connect() should be called and you can trigger the reconnection from there.
